src/kde/kde.t.hh

```diff
@@ -0,0 +1,356 @@
+//--------------------------------------------*-C++-*---------------------------------------------//
+/*!
+ * \file   kde/kde.t.hh
+ * \author Mathew Cleveland
+ * \date   November 10th 2020
+ * \brief  Templated KDE functions for various dimensions and coordinate
+ * systems
+ * \note   Copyright (C) 2018-2020 Triad National Security, LLC.
+ *         All rights reserved. */
+//------------------------------------------------------------------------------------------------//
+
+// clang-format off
+
+#ifndef kde_kde_t_hh
+#define kde_kde_t_hh
+
+#include "kde.hh"
+#include <iostream>
+#include <numeric>
+#include <math.h>
+
+namespace kde {
+
+/*!
+ * reconstruction 
+ * \brief
+ *
+ * Cartesian geometry reconstruction of a 1D distribution
+ *
+ * \param[in] distribution 
+ * \param[in] position
+ * \param[in] band_width
+ * \param[inout] final_distribution returned final function distribution
+ *
+ * Test of kde.
+ */ 
+template<>
+template<>
+std::vector<double> kde<kde_coordinates::CART>::reconstruction<1>(const
```

Comment:
This looks like a fully specialized function.  It should probably live in a `.cc` file instead of a `.t.hh` file.